File: 035/SearchInsertPosition.kt
class SearchInsertPosition {
    fun searchInsert(nums: IntArray, target: Int): Int {
        var left = 0
        var right = nums.size - 1
        while (left <= right) {
            var mid = (left + right) / 2
            if (nums[mid] == target)
                return mid
            else if (nums[mid] < target)
                left = mid + 1
            else
                right = mid - 1
        }
        return left
    }
}

fun main(args: Array<String>) {
    val nums = intArrayOf(1, 3, 5, 6)
    val target = 5
    println(SearchInsertPosition().searchInsert(nums, target))
}

File: 035/SearchInsertPosition.scala
object SearchInsertPosition {
  def searchInsert(nums: Array[Int], target: Int): Int = {
    var i: Int = 0
    while (i < nums.length && nums(i) <= target) {
      if (nums(i) == target) {
        return i
      }
      i += 1
    }
    return i
  }

  def main(args: Array[String]): Unit = {
    val nums = Array(1, 3, 5, 6)
    val target = 5
    println(searchInsert(nums, target))
  }

}
